app/gui/view/component-browser/breadcrumbs/src/lib.rs

```diff
@@ -0,0 +1,485 @@
+//! Breadcrumbs of the Component Browser. It displays a stack of entered module names.
+//!
+//! To learn more about the Component Browser and its components, see the [Component Browser
+//! Design Document](https://github.com/enso-org/design/blob/e6cffec2dd6d16688164f04a4ef0d9dff998c3e7/epics/component-browser/design.md).
+//!
+//! The Breadcrumbs component is displayed as a horizontal list composed of individual breadcrumb
+//! entries (simple text labels) separated by [`entry::Model::Separator`] icons and an optional
+//! [`entry::Model::Ellipsis`] icon at the end of the list. The ellipsis icon shows that the last
+//! module has ancestors and can be further expanded.
+//!
+//! The user can select breadcrumbs either by mouse or by keyboard. After switching to a
+//! higher-level breadcrumb, the lower-level breadcrumbs should become grayed out, letting the
+//! user switch back fast.
+//!
+//! The implementation is based on the [`grid_view::GridView`] with a single row and a variable
+//! number of columns. A custom entry type for the Grid View is implemented in the [`entry`]
+//! module. Each entry has three different representations: a text label, a separator icon, and an
+//! ellipsis icon, and it can switch between these representations if needed.
+
+#![recursion_limit = "1024"]
+// === Features ===
+#![feature(option_result_contains)]
+#![feature(derive_default_enum)]
+#![feature(trait_alias)]
+#![feature(hash_drain_filter)]
+#![feature(bool_to_option)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+// === Non-Standard Linter Configuration ===
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use ensogl_core::prelude::*;
+
+use enso_frp as frp;
+use ensogl_core::application::command::FrpNetworkProvider;
+use ensogl_core::application::frp::API;
+use ensogl_core::application::shortcut::Shortcut;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::layer::Layer;
+use ensogl_core::display::shape::StyleWatchFrp;
+use ensogl_core::gui::Widget;
+use ensogl_core::Animation;
+use ensogl_grid_view as grid_view;
+use ensogl_grid_view::Viewport;
+use ensogl_hardcoded_theme::application::component_browser::searcher::list_panel::breadcrumbs as theme;
+use entry::Entry;
+use grid_view::Col;
+
+
+
+mod entry;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// We virtually divide the breadcrumbs into two regions – left and right. This constant
+/// determines the fraction of the left one.
```

Comment:
The word "fraction" is not explained. What does fraction in this context mean?

Reply:
I rephrased a little by using the word `proportion` instead.